File: archetype-scala/src/main/resources/archetype-resources/src/main/scala/resources/HelloResource.scala
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.resources

import javax.ws.rs.{Produces, QueryParam, GET, Path}
import org.springframework.stereotype.Component
import javax.inject.{Named, Inject}
import com.netflix.config.scala.{DynamicStringProperty => DString}
import ${package}.core.Hello
import javax.ws.rs.core.MediaType
import com.yammer.metrics.annotation.Timed

@Component
@Path("/hello")
class HelloResource @Inject() (@Named("helloText") helloText: DString) {

  @GET
  @Path("/text")
  @Produces(Array(MediaType.TEXT_PLAIN))
  @Timed
  def hellotext(@QueryParam("more") more: Option[String]) =
    helloText.get() + more.getOrElse("")

  @GET
  @Produces(Array(MediaType.APPLICATION_JSON))
  @Timed
  def hellojson(@QueryParam("more") more: Option[String]) =
    new Hello(helloText.get(), "Scala Resource"+more.getOrElse(""))
}

File: archetype-java/src/main/resources/archetype-resources/src/main/java/resources/HelloResource.java
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.resources;

import com.google.common.base.Optional;
import com.yammer.metrics.annotation.Timed;
import org.springframework.stereotype.Component;
import ${package}.core.Hello;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

@Component
@Path("/hello")
public class HelloResource {

    @Inject
    Sayer sayer;

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    @Timed
    public Hello hello(@QueryParam("more") Optional<String> more) {
        return new Hello(sayer.hello(), "Resource"+more.or(""));
    }
}
